tests/flot.annotations.Test.js

```diff
@@ -124,6 +127,49 @@ describe('Flot annotations', function () {
         jasmine.clock().tick(20);
         expect(spy1).toHaveBeenCalledTimes(2);
     });
+    it('should be possible to position the annotation relative to any of the axes when having multiple ones', function () {
+        plot = $.plot(placeholder, [
+            { data: d2, xaxis: 1, yaxis: 1 },
+            { data: d3, xaxis: 2, yaxis: 2 }
+        ],
+        {
+            annotations: [
+                {
+                    show: true,
+                    location: 'absolute',
+                    x: 7,
+                    y: 12,
+                    label: 'hello world2<br>newline',
+                    arrowDirection: 'n',
+                    showArrow: true,
+                    defaultxaxis: 2,
+                    defaultyaxis: 2
+                }
+            ],
+            xaxes: [
+                { position: 'bottom' },
+                { position: 'top' }
+            ],
+            yaxes: [
+                { position: 'left' },
+                { position: 'right' }
+            ]
+        });
+
+        jasmine.clock().tick(20);
+
+        var pos1 = plot.p2c({
+            x2: 7,
+            y2: 12
+        });
+
+        var expectedX1 = pos1.left;
+        var expectedY1 = pos1.top;
+        var spy1 = spyOn(ctx, 'fillRect').and.callThrough();
+        plot.triggerRedrawOverlay();
+        jasmine.clock().tick(20);
+        expect(spy1).not.toHaveBeenCalledWith(expectedX1, expectedY1, jasmine.any(Number), jasmine.any(Number));
```

Comment:
I don't think this test clearly tests what we are trying to describe here. I believe a proper test would explicitly have two sets of axes with different ranges (which is implicit in this test...driven by the data autoscaling the axes), and then two annotations assigned to a different axis-pair, such that (for ease of testing) they overlapped exactly in the plot area. The test would then make sure that we called the same 'fillRect' method twice with the same values despite them being positioned to two different sets of domain values.